gcore/gdaldriver.cpp

```diff
@@ -949,16 +960,56 @@ GDALDataset *GDALDriver::CreateCopy( const char * pszFilename,
     if( pfnProgress == nullptr )
         pfnProgress = GDALDummyProgress;
 
+/* -------------------------------------------------------------------- */
+/*      If no INTERLEAVE creation option is given, we will try to add   */
+/*      one that matches the current srcDS interleaving                 */
+/* -------------------------------------------------------------------- */
+    char** papszOptionsSmart = 0;
+    const bool bHasInterleaveCreationOptions =
+         (CSLFetchNameValue(papszOptions, "INTERLEAVE") != 0);
+    if (!bHasInterleaveCreationOptions && poSrcDS)
+    {
+        const char* srcInterleave = poSrcDS->GetMetadataItem("INTERLEAVE", "IMAGE_STRUCTURE");
+        const bool isENVIDriver = !strcmpi(this->GetDescription(), "ENVI");
```

Comment:
replace !strcmpi() uses by EQUAL(), which is our portability macro

Reply:
I'm a big unconfortable with explicitly testing for the ENVI driver. It lacks generality. My thoughts would have rather to parse the XML of GDAL_DMD_CREATIONOPTIONLIST metadata item of the target driver, see if it has a INTERLEAVE option, and what its values are.  If they are PIXEL/BAND  or BSQ/BIP/BIL, you can do the remapping. Otherwise you give up (potentially with a CPLDebug() message so that we know either the driver options must be standardized, or we have to add a new case)